go/pkg/zen/middleware_metrics.go

```diff
@@ -77,7 +77,8 @@ func WithMetrics(eventBuffer EventBuffer) Middleware {
 				ipAddress = ips[0]
 			}
 
-			if s.r.Header.Get("X-Unkey-Metrics") != "disabled" {
+			// Check both the header-based control and session-based control
+			if s.r.Header.Get("X-Unkey-Metrics") != "disabled" && s.ShouldLog() {
```

Comment:
It might be cleaner to move the header check into the few routes that need it instead of having this global middleware pollution.

@coderabbitai please create an issue. The relevant routes are liveness, the chproxy ones and the ratelimit.limit I believe